src/features/instance/operations/queries/getRegistrationInfo.ts

```diff
@@ -0,0 +1,26 @@
+import instanceClient from '@/config/instanceClient';
+
+import { queryOptions } from '@tanstack/react-query';
+
+type RegistrationInfoResponse = {
+	license_expiration_date: string;
+	ram_allocation: number;
+	registered: boolean;
+	version: string;
+};
+
+function getRegistrationInfoQueryOptions(instanceId: string) {
+	return queryOptions({
+		queryKey: [instanceId, 'registration_info'] as const,
+		queryFn: async () => {
```

Comment:
Since this function doesn't reference `this`, I don't see the point in doing an anonymous function.

```typescript
		async queryFn() {
			const { data } = await instanceClient.post('/', {
				operation: 'registration_info',
			});
			return data as RegistrationInfoResponse;
		},
```

Reply:
If i'm not mistaken this is a recommended ways for an async function to work inside `queryFn()` I'll double check!